bool reset_left
bool reset_back
bool reset_right